### PR TITLE
aarch64: ZHF for aarch64 (1/??)

### PR DESCRIPTION
**Pull request by samueldr**
###### Motivation for this change

Reduce the amount of failures for aarch64. This is done mainly through disabling builds for packages **which cannot** build on aarch64. I have skipped over some packages which could realistically be built, with effort. Most of those disabled here use x86_64 features pretty explicitly.

There is either a comment preceding the `badPlatforms` property with details, or a detailed explanation in the commit message.

* * *

###### Some additional notes:

**aerospike** has been disabled again on aarch64-linux, this time it's been removed as a default for aarch64 in the module, making it not fail uselessly in Hydra. ~~I'm thinking the entire module could also be conditional on valid platforms for the package, but it would stop end-users from using it with a fork for fixing it :/.~~ (With `defaultText` it's not as much an issue. No hairy conditional.) cc @kalbasit 

**boost** 1.55 was previously flagged as not building on aarch64-linux for versions preceding 1.59, but the semantics of *platforms* changed, re-adding the failing build.

**openbabel** is given a patch from upstream to fix its build. This additional patch will be removed once they release another version including the change.

Anything else is trivial.

###### Things done

- ✔️ Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - ✔️ NixOS
   - ⬜ macOS
   - ⬜ other Linux distributions
- ⬜ Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- ⬜ Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- ⬜ Tested execution of all binary files (usually in `./result/bin/`)
- ⬜ Determined the impact on package closure size (by running `nix path-info -S` before and after)
- ✔️ Assured whether relevant documentation is up to date
- ✔️ Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/databases/aerospike.nix [hunk 1]
```diff
@@ -42,7 +42,11 @@ in
       enable = mkEnableOption "Aerospike server";
 
       package = mkOption {
-        default = pkgs.aerospike;
+        default =
```
Comment (pbogdan) [inline nixos/modules/services/databases/aerospike.nix]: Wouldn't specifying `defaultText` be a better solution? Then no one would need to build `aerospike` to have access to the NixOS manual.

File: pkgs/development/libraries/beignet/default.nix [hunk 2]
```diff
@@ -107,5 +107,7 @@ stdenv.mkDerivation rec {
     license = licenses.lgpl21Plus;
     maintainers = with maintainers; [ artuuge zimbatm ];
     platforms = platforms.linux;
+    # Requires libdrm_intel
+    badPlatforms = [ "aarch64-linux" ];
```
Comment (Mic92) [inline pkgs/development/libraries/beignet/default.nix]: What is the semantics of badPlatforms?

Comment (vcunat) [reply pkgs/development/libraries/beignet/default.nix]: Right, I don't that described or even affecting any processing in nixpkgs.

Comment (samueldr) [reply pkgs/development/libraries/beignet/default.nix]: See b0482248fefbf3b6cdd9c92053cfb49778a3a3a8, platforms used are, simplified, `platforms - badplatforms`.

For beignet specifically, I would have preferred having a way to compose `linux × [i686, x86_64]`, but I don't think it's possible with how the platforms work currently.

File: nixos/modules/services/databases/aerospike.nix (hunk 1 above)
Comment (samueldr) [reply nixos/modules/services/databases/aerospike.nix]: Right! I knew there was a way, but completely forgot what it was. Thanks!

File: pkgs/development/libraries/beignet/default.nix (hunk 2 above)
Comment (samueldr) [reply pkgs/development/libraries/beignet/default.nix]: AFAIK this is the last remaining open "issue" about this PR. Anything required to change about the use of `badPlatforms`? Any better way to semantically say "x86-based platforms only" while still allowing to say "and linux" or "and either linux or darwin"?

**Pull request merged**